code/modules/admin/verbs/buildmode.dm

```diff
@@ -354,10 +359,16 @@
 						if(GLOB.reloading_map)
 							to_chat(user, "<span class='boldwarning'>You are already reloading an area! Please wait for it to fully finish loading before trying to load another!</span>")
 							return
-					G.defineRegion(cornerA,cornerB,1)
-					G.generate()
+					G.defineRegion(cornerA, cornerB, 1)
+					for(var/t in G.map)
+						preview += image('icons/turf/overlays.dmi', t ,"greenOverlay")
+					user.client.images += preview
+					var/confirm = alert("Are you sure you want run the map generator?", "Run generator", "Yes", "No")
+					if(confirm == "Yes")
+    					G.generate()
 					cornerA = null
 					cornerB = null
+					usr.client.images -= preview
```

Comment:
you should empty the list afterward, since you're keeping it on the buildmode object.
otherwise you'll end up with many copies of the images.

Reply:
fixed, tyvm